user-service/controller/user-controller.js

```diff
@@ -37,6 +37,61 @@ export async function createUser(req, res) {
   }
 }
 
+// Password complexity check
+// This is the second check for password complexity
+// The first check is done in the frontend
+const isPasswordComplex = (password) => {
+  const minLength = 8;
+  const hasUpperCase = /[A-Z]/.test(password);
+  const hasSpecialChar = /[!@#$%^&*()_+\-=\[\]{};':"\\|,.<>\/?]+/.test(
+    password
+  );
+
+  return password.length >= minLength && hasUpperCase && hasSpecialChar;
+};
```

Comment:
hmm is it common practice to check in backend also? i see the point but wondering if it's normal